subsys/net/ip/net_if.c

```diff
@@ -157,7 +164,27 @@ static bool net_if_tx(struct net_if *iface, struct net_pkt *pkt)
 			net_pkt_set_queued(pkt, false);
 		}
 
+#if defined(CONFIG_NET_CONTEXT_TIMESTAMP)
+		if (context) {
+			if (net_context_get_timestamp(context, pkt,
+						      &start_timestamp) < 0) {
+				start_timestamp.nanosecond = 0;
+			} else {
+				pkt_priority = net_pkt_priority(pkt);
```

Comment:
And what happens on the other branch? pkt_priority stays uninitialized?

Reply:
That branch is not entered in that case as start_timestamp.nanosecond == 0